### PR TITLE
DCMAW-9756: Fixes in credential handler following creation of AWS accounts

### PR DESCRIPTION
**Pull request by jmooney-dd**
​DCMAW-9756

### What changed
AWS accounts have now been provisioned and a number of infrastructure changes have been made to ensure lambdas can be executed successfully

## Checklists

- [x] There is a ticket raised for this PR that is present in the branch name
- [x] No PII data logged. [See guidance here](https://govukverify.atlassian.net/wiki/spaces/DCMAW/pages/3502407722/PII+Logging+Considerations)
- [ ] Demo to a BA, TA, and the team.
- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

File: backend-api/template.yaml [hunk 1]
```diff
@@ -315,7 +316,25 @@ Resources:
...
+        - PolicyName: DynamoReadSessionsTable
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - dynamodb:GetItem
+                  - dynamodb:Query
+                Resource: !GetAtt SessionsTable.Arn
+        - PolicyName: AsyncTokenFunctionSecretsManagerPolicy
```
Comment (sandymay-dd) [inline backend-api/template.yaml:328]:
```suggestion
        - PolicyName: AsyncCredentialFunctionSecretsManagerPolicy
```

File: backend-api/template.yaml [hunk 2]
```diff
@@ -315,7 +316,25 @@ Resources:
...
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - secretsmanager:GetSecretValue
+                Resource:
+                  - !Sub arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:${Environment}/clientRegistry-?????? # wildcard as AWS automatically appends 6 characters to the end of a secret arn
+        - PolicyName: AsyncTokenFunctionLoggingPolicy
```
Comment (sandymay-dd) [inline backend-api/template.yaml:337]:
```suggestion
        - PolicyName: AsyncCredentialFunctionLoggingPolicy
```

File: backend-api/template.yaml [hunk 3]
```diff
@@ -325,19 +344,69 @@ Resources:
                   - logs:CreateLogStream
                   - logs:PutLogEvents
                 Resource: arn:aws:logs:*:*:*
+        - PolicyName: AsyncTokenFunctionKmsPolicy
```
Comment (sandymay-dd) [inline backend-api/template.yaml:347]:
```suggestion
        - PolicyName: AsyncCredentialFunctionKmsPolicy
```

File: backend-api/template.yaml [hunk 4]
```diff
@@ -325,19 +344,69 @@ Resources:
...
+            Version: "2012-10-17"
+            Statement:
               - Effect: Allow
                 Action:
-                  - secretsmanager:GetSecretValue
-                Resource:
-                  - !Sub arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:${Environment}/clientRegistry-?????? # wildcard as AWS automatically appends 6 characters to the end of a secret arn
+                  - kms:Verify
+                Resource: !GetAtt KMSSigningKey.Arn
+        - PolicyName: AsyncTokenFunctionSQSPolicy
```
Comment (sandymay-dd) [inline backend-api/template.yaml:355]:
```suggestion
        - PolicyName: AsyncCredentialFunctionSQSPolicy
```

File: backend-api/template.yaml [hunk 5]
```diff
@@ -315,7 +316,25 @@ Resources:
...
       Policies:
-        - PolicyName: AsyncTokenFunctionSsmPolicy
+        - PolicyName: DynamoReadSessionsTable
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - dynamodb:GetItem
+                  - dynamodb:Query
```
Comment (sandymay-dd) [inline backend-api/template.yaml:326]: We haven't actually tested this yet. What Dynamo commands do we run? I think it's just GetItem and PutItem from memory? 

Please may you check and update these - no need to test as we're blocked by KMS

File: backend-api/template.yaml (hunk 3 above)
Comment (jmooney-dd) [reply backend-api/template.yaml:347]: Update backend-api/template.yaml policy name

File: backend-api/template.yaml (hunk 5 above)
Comment (jmooney-dd) [reply backend-api/template.yaml]: Updated

**Review by sandymay-dd — DISMISSED** [review]

**Review by sandymay-dd — APPROVED** [review]

**Pull request merged**